EIPS/eip-4337.md

```diff
@@ -349,12 +416,30 @@ During bundling, the client should:
 
 After creating the batch, before including the transaction in a block, the client should:
 
-* Run `eth_estimateGas` with maximum possible gas, to verify the entire `handleOps` batch transaction, and use the estimated gas for the actual transaction execution.
-* If the call reverted, check the `FailedOp` event. A `FailedOp` during `handleOps` simulation is an unexpected event since it was supposed to be caught by the single-UserOperation simulation. Remove the failed op that caused the revert from the batch and drop from the mempool.
-  If the error is caused by a factory (error code is "AA1.") or paymaster (error code is "AA3."), then also drop from mempool all other UserOps of this entity.
-  Repeat until `eth_estimateGas` succeeds.
-
-In practice, restrictions (2) and (3) basically mean that the only external accesses that the account and the paymaster can make are reading code of other contracts if their code is guaranteed to be immutable (eg. this is useful for calling or delegatecalling to libraries).
+* Run `debug_traceCall` with maximum possible gas, to enforce the validation opcode and precompile banning and storage access rules,
+  as well as to verify the entire `handleOps` batch transaction,
+  and use the consumed gas for the actual transaction execution.
+* If the call reverted, check the `FailedOp` event.
+  A `FailedOp` during `handleOps` simulation is an unexpected event since it was supposed to be caught
+  by the single-UserOperation simulation.
```

Comment:
Therefore, remove the offending UserOperation from the bundle.  (we elaborate on how to deal with the offender but I think we neglected to mention the obvious - that the UserOperation itself is removed from the bundle)